src/bgw/scheduler.c

```diff
@@ -548,10 +553,18 @@ cmp_next_start(const void *left, const void *right)
 static void
 start_scheduled_jobs(register_background_worker_callback_type bgw_register)
 {
+	List *ordered_scheduled_jobs;
 	ListCell *lc;
 	Assert(CurrentMemoryContext == scratch_mctx);
+
 	/* Order jobs by increasing next_start */
-	List *ordered_scheduled_jobs = list_qsort(scheduled_jobs, cmp_next_start);
+#if PG13_LT
+	ordered_scheduled_jobs = list_qsort(scheduled_jobs, cmp_next_start);
+#else
+	/* PG13 does in-place sort */
+	ordered_scheduled_jobs = scheduled_jobs;
+	list_sort(ordered_scheduled_jobs, cmp_next_start);
+#endif
```

Comment:
This can be encapsulated into an ordering function, which will always produce a sorted copy. E.g.:
```c
static List *
list_sort_with_copy(List *origin) 
{
	List *new_list;

#if PG13_LT
	new_list = list_qsort(origin, cmp_next_start_void);
#else
	/* PG13 does in-place sort */
	new_list = list_copy(origin);
	list_sort(new_list, cmp_next_start);
#endif
}
```
Then:
```suggestion
	ordered_scheduled_jobs = order_list_with_copy(scheduled_jobs);
```